cpp/src/copying/get_element.cu

```diff
@@ -174,11 +176,18 @@ struct get_element_functor {
                                                    mr);
   }
 
-  template <typename T, typename... Args>
-  std::enable_if_t<std::is_same<T, struct_view>::value, std::unique_ptr<scalar>> operator()(
-    Args &&...)
+  template <typename T, std::enable_if_t<std::is_same<T, struct_view>::value> *p = nullptr>
+  std::unique_ptr<scalar> operator()(
+    column_view const &input,
+    size_type index,
+    rmm::cuda_stream_view stream,
+    rmm::mr::device_memory_resource *mr = rmm::mr::get_current_device_resource())
   {
-    CUDF_FAIL("get_element_functor not supported for struct_view");
+    bool valid = is_element_valid_sync(input, index, stream);
+    auto row_contents =
+      std::make_unique<column>(slice(input, index, index + 1), stream, mr)->release();
+    auto scalar_contents = std::make_unique<table>(std::move(row_contents.children));
+    return std::make_unique<struct_scalar>(std::move(*scalar_contents), valid, stream, mr);
```

Comment:
I think `std::move(*scalar_contents)`  is wrong as the `unique_ptr` hasn't been informed that it doesn't have ownership anymore. I think you want:

```cpp
auto scalar_contents = table(std::move(row_contents.children));
std::make_unique<struct_scalar>(std::move(scalar_contents), valid, stream, mr);
```

Reply:
> `std::make_unique<struct_scalar>(std::move(scalar_contents), valid, stream, mr);`

Not quite. The `struct_scalar` constructor needs a `table&&`.

> I think std::move(*scalar_contents) is wrong as the unique_ptr hasn't been informed that it doesn't have ownership anymore.

The `unique_ptr` retains ownership of the "moved-from" object, which should be safe to destroy when the `unique_ptr` goes out of scope.

Reply:
I believe you are correct.  To clarify, the object (the table) being pointed to by the `std::unique_ptr<>` has had it's internal buffers emptied out, but the table held by the pointer itself still exists, so when it gets deleted, it's destructor does nothing. 

It is a little bit subtle though. @robertmaynard 's suggestion makes the sequence of operations a lot more clear.

As a side note, it's a little surprising the PR that added these move constructors didn't actually have any tests for them.

Reply:
Ah, I misread @robertmaynard's suggestion: `scalar_contents` is a `table`, in his case. Yes, his way reads better.

Reply:
Thanks for the comments. Moving to @robertmaynard 's suggestion.

From a lower level perspective, previously
```
std::make_unique<struct_scalar>(std::move(*scalar_contents), valid, stream, mr);
```
creates a `table` object with the contents that was pointed to by the `unique_ptr` and was later moved to the `struct_scalar`. After this step, the `unique_ptr` still points to a `table` object, but as a result of default move constructor, that object now contains 0 columns, effectively empty. It's safe for `unique_ptr` to deallocate this object because the `table` object is still live, its internal `_columns` field is also live, only it's empty.

Agreed it's subtle compared to the new writings.